### PR TITLE
feat: add highlights to list pages

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR adds support for viewing hihglights of contributors in a list and filtering based on contributors login, range of days since the highlight was posted and repos tagged in the highlight.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes https://github.com/open-sauced/app/issues/2069

## Mobile & Desktop Screenshots/Recordings

Uploading Screen Recording 2023-12-12 at 10.04.18.mov…

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
Contributor type and sort by filters are yet to be added, but this is good to ship for first iteration to get user's feedback before added other filters. 
image

**Review by jpmcb — COMMENTED** [review]
This is amazing! Well done!!

---

I noticed this inconsistency with the scrollbar on the right. Here's without the scrollbar (since it's only my highlight and it's not long):

Screenshot 2023-12-12 at 9 28 35 AM

and here's with the scrollbar (since there are alot of items in the list):

Screenshot 2023-12-12 at 9 28 45 AM

this causes a shift in the UI slightly to the left when using the search for contributors (or doing other filtering).

This seems to be related to:
- https://github.com/open-sauced/app/issues/2252

File: components/atoms/Radio/radio-group.tsx
Comment (nickytonline) [inline components/atoms/Radio/radio-group.tsx:29]: Refactor (blocking): Remove the `focus-visible` classes from the `<RadioGroupPrimitive.Item />` component.

```suggestion
    <RadioGroupPrimitive.Item
      ref={ref}
      className={clsx(
        "aspect-square h-4 w-4 rounded-full border border-primary text-sauced-orange shadow disabled:cursor-not-allowed disabled:opacity-50",
        className
      )}
      {...props}
    >
```

Add add a selector to the globals.css file for global focus styles for the radio buttons. Radix radiot buttons are buttons with `role="radio"`.

```diff
 input[type="checkbox"]:focus-visible,
- [role="checkbox"]:focus-visible {
+ [role="checkbox"]:focus-visible,
+ [role="radio"]:focus-visible {
   @apply outline-none;
   @apply ring;
   @apply border-orange-500; 
   @apply ring-orange-100;
 }
```

File: pages/lists/[listId]/highlights.tsx
Comment (nickytonline) [inline pages/lists/[listId]/highlights.tsx:148]: The `<SingleSelect />` component needs a focus state. I'm going to look at adding this to the global styles as we need it elsewhere.

CleanShot 2023-12-12 at 16 43 02

File: pages/lists/[listId]/highlights.tsx
Comment (nickytonline) [inline pages/lists/[listId]/highlights.tsx:193]: Instead of this message, consider adding the time range as there are highlights in some cases depending on the time range given.

CleanShot 2023-12-12 at 16 44 38

**Review by nickytonline — CHANGES_REQUESTED** [review]
This is looking good Sunday. A couple change requests.

File: lib/hooks/useList.ts
Comment (nickytonline) [inline lib/hooks/useList.ts:182]: Any reason `limit=50` is hard-coded?

**Review by nickytonline — CHANGES_REQUESTED** [review]

File: lib/hooks/useList.ts
Comment (OgDev-01) [reply lib/hooks/useList.ts:119]: This is a pattern that has been in the code base for sometime now... we might want to consider addressing this in a seperate PR since it is in other areas of the app.

File: lib/hooks/useList.ts
Comment (OgDev-01) [reply lib/hooks/useList.ts:182]: We don't have any UI provisions to update this limit programatically ATM, so i assumed setting to a limit of 50 will be a goo start for now.

Comment (OgDev-01) [reply lib/hooks/useList.ts:182]: We don't have any UI provisions to update this limit programatically ATM, so i assumed setting to a limit of 50 will be a goo start for now.

File: lib/hooks/useList.ts
Comment (OgDev-01) [reply lib/hooks/useList.ts:119]: This is a pattern that has been in the code base for sometime now... we might want to consider addressing this in a seperate PR since it is in other areas of the app.

File: lib/hooks/useList.ts
Comment (nickytonline) [reply lib/hooks/useList.ts:182]: Cool cool. We can iterate on it.

File: pages/lists/[listId]/highlights.tsx
Comment (nickytonline) [reply pages/lists/[listId]/highlights.tsx:148]: This work is merged for focus states. #2319

File: CHANGELOG.md
Comment (nickytonline) [inline CHANGELOG.md:8]: Looks like there was a merge from beta issue. This file shouldn't be a part of the PR.

File: pages/lists/[listId]/highlights.tsx
Comment (nickytonline) [inline pages/lists/[listId]/highlights.tsx:133]: ux (blocking): smaller screens hide the search/filtering for highlighting. It should probably appear at the top for now, but we could iterate on better UX for this for small screens.

CleanShot 2023-12-14 at 07 02 46

**Review by nickytonline — CHANGES_REQUESTED** [review]

**Comment by OgDev-01** [comment]
filter has been updated to show on mobile view now @nickytonline 


https://github.com/open-sauced/app/assets/62995161/826468e5-61dc-4a0e-9e9d-467a5863f2f6

**Comment by nickytonline** [comment]
When there are no highlights and then you go to a day range that does have highlights, the page shifts.

If we keep the empty highlights state the same size as the cards it should prevent the shift.


https://github.com/open-sauced/app/assets/833231/f218879d-7212-4193-9fcd-0df002acd22d

**Comment by nickytonline** [comment]
We should also add paging like we do on the main highlights feed.

CleanShot 2023-12-19 at 09 31 37

File: lib/hooks/search-users.ts
Comment (brandonroberts) [inline lib/hooks/search-users.ts:1]: This needs to be reverted. We switched to use the GitHub search API for more flexibility

#2331

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: components/organisms/SearchDialog/search-dialog.tsx
Comment (brandonroberts) [inline components/organisms/SearchDialog/search-dialog.tsx:66]:
```suggestion
      const data = await searchUsers(debouncedSearchTerm, providerToken);
```

File: lib/utils/github.ts
Comment (brandonroberts) [inline lib/utils/github.ts:22]: Is this being used?

File: next-types.d.ts
Comment (brandonroberts) [inline next-types.d.ts:347]: Not being used

File: package.json
Comment (brandonroberts) [inline package.json:132]: Why was this removed?

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: package.json
Comment (OgDev-01) [reply package.json:132]: not sure why this change... will revert

**Comment by OgDev-01** [comment]
Paging is included here https://github.com/open-sauced/insights/blob/ogdev/2069-list-highlights/pages/lists/%5BlistId%5D/highlights.tsx @nickytonline

**Comment by nickytonline** [comment]
There is still the shifiting of the content that [I mentioned previously](https://github.com/open-sauced/app/pull/2194#issuecomment-1862866267) running the latest deploy preview.

https://github.com/open-sauced/app/assets/833231/5c309d99-ae60-474c-8ec2-708b9610996b

**Comment by bdougie** [comment]
@OgDev-01 & @isabensusan what's the reasoning for only looking back 30 days?

Could I suggest we remove the data filter in favor of the repo filter? Also, add pagination?

**Review by bdougie — CHANGES_REQUESTED** [review]
Replace date filter for the repo filter. The date filter is really not useful in this context, but the repo filter would provide more value.

**Comment by isabensusan** [comment]
@bdougie the date range filter was meant to help users navigate long lists of highlights if they wanted to, but I agree it isn't really a necessity and the way it's set up right now (with "Past 30 days" pre-selected) is bad UX, so let's get rid of it.

Just to do a quick summary of the changes requested:
### 1. Repository filter: 
we can bring over the repo filter component from the higlights page for now:
image

And ideally in a second iteration, I'd love to have something like a searchable multi-select dropdown:
image

### 2. Add pagination
The same way we do with the Highlights feed
image

And a style fix:

### 3. Make tab component line span the whole width
Right now it only goes half way across the container width, making it look broken.
image

Can we make it take up the whole container?
image

**Comment by nickytonline** [comment]
> @bdougie the date range filter was meant to help users navigate long lists of highlights if they wanted to, but I agree it isn't really a necessity and the way it's set up right now (with "Past 30 days" pre-selected) is bad UX, so let's get rid of it.
> 
> Just to do a quick summary of the changes requested:
> 
> ### 1. Repository filter:
> we can bring over the repo filter component from the higlights page for now: image
> 
> And ideally in a second iteration, I'd love to have something like a searchable multi-select dropdown: image
> 
> ### 2. Add pagination
> The same way we do with the Highlights feed image
> 
> And a style fix:
> 
> ### 3. Make tab component line span the whole width
> Right now it only goes half way across the container width, making it look broken. image
> 
> Can we make it take up the whole container? image

Sounds good to me.

**Comment by OgDev-01** [comment]
Pagination is currently broken in the api for this endpoint... workin on that in https://github.com/open-sauced/api/issues/468

**Comment by OgDev-01** [comment]
pagination fixed 🤌 
image

**Comment by nickytonline** [comment]
When I go to the highlights of one of my lists, all I see is this. Also, the repo filter appears to be missing.

CleanShot 2024-01-08 at 15 32 59

I went to another and see some highlights now, but still no repo filter.

**Comment by OgDev-01** [comment]
> When I go to the highlights of one of my lists, all I see is this.

This is because no contributor in the list has created any highlight in the last 30Days

**Comment by bdougie** [comment]
> This is because no contributor in the list has created any highlight in the last 30Days

Can we disable the search if there are no highlights?

**Comment by OgDev-01** [comment]
> > This is because no contributor in the list has created any highlight in the last 30Days
> 
> Can we disable the search if there are no highlights?

That shouldn't be a heavy lift... will do that today

**Comment by OgDev-01** [comment]
> Can we disable the search if there are no highlights?

This has been implemented @bdougie... this will be ready for a second review once https://github.com/open-sauced/api/pull/473 is merged

**Comment by OgDev-01** [comment]
All Good on this and ready for a second review
image
cc: @open-sauced/engineering and @isabensusan

**Review by jpmcb — COMMENTED** [review]
The "Lists" tab on the hub seems broken in this for me: it redirects to the highlights home:

```
https://deploy-preview-2194--oss-insights.netlify.app/feed?redirectedFrom=%2Fhub%2Flists
```

**Comment by nickytonline** [comment]
> The "Lists" tab on the hub seems broken in this for me: it redirects to the highlights home:
> 
> ```
> https://deploy-preview-2194--oss-insights.netlify.app/feed?redirectedFrom=%2Fhub%2Flists
> ```

Odd. I'm able to hit `/hub/lists` no problem when logging in. 🤔

**Comment by bdougie** [comment]
> > Can we disable the search if there are no highlights?
> 
> This has been implemented @bdougie... this will be ready for a second review once [open-sauced/api#473](https://github.com/open-sauced/api/pull/473) is merged

Search is disabled, but my expectation is the cursor is disabled on hover as well.

`cursor-not-allowed`

Screen Shot 2024-01-09 at 9 04 24 PM

**Comment by OgDev-01** [comment]
> The "Lists" tab on the hub seems broken in this for me: it redirects to the highlights home:

Are you logged in?

**Review by jpmcb — COMMENTED** [review]
Spacing looks off:

Screenshot 2024-01-10 at 8 51 41 AM

But looks ok in a responsive layout:

Screenshot 2024-01-10 at 8 52 53 AM

if I make my browser window smaller, I can engage the right styling:

Screenshot 2024-01-10 at 8 53 40 AM

**Comment by OgDev-01** [comment]
> Spacing looks off:

took off a redundant class in the layout with my latest push, and I think that fixes the issue.

**Review by nickytonline — CHANGES_REQUESTED** [review]
There's some weird behaviour still. As well, the page still shifts if there are no results.

Filtering on repos works, but as soon as I start typing, if cuts me off from typing, displays no search results and I can no longer type in the search input unless I click the X to clear it.

https://github.com/open-sauced/app/assets/833231/3f8bccbe-884a-47b1-b2f7-f975423f06d3

**Comment by OgDev-01** [comment]
The issue with search is fixed. I will open a papercut issue for the layout shifting so we can get this merged as it's been around for a while

**Comment by nickytonline** [comment]
The search fix looks good. There's a bit of jank as you search because it searches, shows skeleton loaders, comes back quickly, but that can be addressed in the follow up issue about the layout shift.

**Comment by nickytonline** [comment]
I thought there was paging or are we showing all highlights for the past 30 days?

**Comment by OgDev-01** [comment]
> I thought there was paging or are we showing all highlights for the past 30 days?

Yes

**Comment by OgDev-01** [comment]
> I thought there was paging or are we showing all highlights for the past 30 days?

You can add me to your list to get more highlights and see the pagination

**Comment by nickytonline** [comment]
I see the paging now. It's a bit weird to go to the next page and then it scrolls automatically all the way back up to the start of the highlights. I think for highlights, it would probably make more sense with a Load More button.

https://github.com/open-sauced/app/assets/833231/619edf7f-234a-4476-a53e-d32bb6671264

We can probably merge this and tweak that after, but I'll let others chime in.

**Comment by brandonroberts** [comment]
I think its good to merge and iterate on

**Review by brandonroberts — APPROVED** [review]

**Review by bdougie — APPROVED** [review]
🚀

**Review by jpmcb — APPROVED** [review]
🫡

**Pull request merged**
